### PR TITLE
Add decoding from root keys doc

### PR DESCRIPTION
**Pull request by tonyd256**

### CODE REVIEW

File: Documentation/Decode-Root-Keys.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+```
+{
+  "user": {
+    "id": 1881372911,
+    "name": "George Sr"
+  }
+}
+```
+
+In this case, you can't use the global `decode` function. The easiest way
```
Comment (gfontenot) [inline Documentation/Decode-Root-Keys.md]: This paragraph reads a little funky to me.

**Comment by gfontenot** [comment]
Imma ping @thoughtbot/copy-editing here as well

File: Documentation/Decode-Root-Keys.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+The easiest way to decode models from json is to use the global `decode`
```
Comment (gabebw) [inline Documentation/Decode-Root-Keys.md]: I'd add a header so people have context even if they're directly linked here. How about `# Decoding Root Keys`?

File: Documentation/Decode-Root-Keys.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+like so:
+
+```swift
+let json = JSON.parse(anyObject)
+
+let user: Decoded<User> = json <| "user"
+// or
+let user: User? = json <| "user"
+```
+
```
Comment (gabebw) [inline Documentation/Decode-Root-Keys.md]: extra line

File: Documentation/Decode-Root-Keys.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
+The easiest way to decode models from json is to use the global `decode`
+function. You can pass the `AnyObject` variable returned from
+`NSJSONSerialization` directly to `decode` and get the model back. However,
+many times the model json is embedded within a root key:
+
+```
+{
+  "user": {
+    "id": 1881372911,
+    "name": "George Sr"
```
Comment (gabebw) [inline Documentation/Decode-Root-Keys.md]: Sr -> Senior

File: Documentation/Decode-Root-Keys.md (hunk 2 above)
Comment (mcmire) [inline Documentation/Decode-Root-Keys.md]: json -> JSON

Comment (mcmire) [inline Documentation/Decode-Root-Keys.md]: Also, what about: 

> The easiest way to decode models from JSON is to use the global `decode` function: pass the `AnyObject` variable returned from `NSJSONSerialization` directly to `decode`, and you'll get the model back.

File: Documentation/Decode-Root-Keys.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+The easiest way to decode models from json is to use the global `decode`
+function. You can pass the `AnyObject` variable returned from
+`NSJSONSerialization` directly to `decode` and get the model back. However,
+many times the model json is embedded within a root key:
```
Comment (mcmire) [inline Documentation/Decode-Root-Keys.md]: model json -> JSON for the model?

File: Documentation/Decode-Root-Keys.md (hunk 1 above)
Comment (mcmire) [inline Documentation/Decode-Root-Keys.md]: The easiest way what?

You've already said "the easiest way" at the beginning of this document, so what about:

> In this case, you can't use the global `decode` function [edit: is there a specific reason why?]. To get around this, first parse the `AnyObject` into `JSON`, then use the `<|` operator to ... [what are you doing here?]

**Comment by tonyd256** [comment]
Thanks for the feedback everyone! This PR will be closed because I need to merge in the documentation branch so master is up to date. I'll reopen with this feedback taken in.

**Pull request closed without merging**